examples/block_writer/main.go

```diff
@@ -0,0 +1,202 @@
+var numBlocks uint64
+
+// A blockWriter writes blocks of random data into cockroach in an infinite
+// loop.
+type blockWriter struct {
+	id   string
+	db   *sql.DB
+	rand *rand.Rand
+}
+
+func newBlockWriter(db *sql.DB) *blockWriter {
+	source := rand.NewSource(int64(time.Now().UnixNano()))
+	return &blockWriter{
+		db:   db,
+		id:   uuid.NewUUID4().String(),
+		rand: rand.New(source),
+	}
+}
+
+// run is an infinite loop in which the blockWriter continuously attempts to
+// write blocks of random data into a table in cockroach DB.
+func (bw *blockWriter) run() {
+	insert := `INSERT INTO blocks (writer_id, block_id, raw_bytes) VALUES ($1, $2, $3)`
+	for {
+		blockID := bw.rand.Int63()
+		blockData := bw.randomBlock()
+		if _, err := bw.db.Exec(insert, bw.id, blockID, blockData); err != nil {
+			if log.V(1) {
+				log.Warning(err)
+			}
+			continue
+		}
+		atomic.AddUint64(&numBlocks, 1)
+	}
+}
+
+// randomBlock generates a slice of randomized bytes. Random data is preferred
+// ot prevent any sort of compression in storage.
+func (bw *blockWriter) randomBlock() []byte {
+	blockSize := bw.rand.Intn(*maxBlockSizeBytes-*minBlockSizeBytes) + *minBlockSizeBytes
+	blockData := make([]byte, blockSize)
+	for i := range blockData {
+		blockData[i] = byte(bw.rand.Int() & 0xff)
+	}
+	return blockData
+}
+
+// parseFlags ensures that values taken from command line flags are valid.
+func parseFlags() {
+	flag.Parse()
+	if *concurrency < 1 {
+		log.Fatal("Value of 'concurrency' flag must be greater than or equal to 1")
+	}
+	if *maxBlockSizeBytes < *minBlockSizeBytes {
+		log.Fatal("Value of 'max-block-bytes' must be greater than or equal to value of 'min-block-bytes'")
+	}
+}
+
+// maybeStartLocalServer will start a temporary local server if no database URL has been
+// provided.
+func maybeStartLocalServer() *server.TestServer {
+	if *dbURL != "" {
+		return nil
+	}
+	// Start a local test server.
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	*dbURL = "https://root@" + serv.ServingAddr() + "?certs=test_certs"
+	return serv
+}
+
+// setupDatabase performs initial setup for the example, creating a database and
+// with a single table. If the desired table already exists on the cluster, the
+// existing table will be dropped.
+func setupDatabase() *sql.DB {
+	var db *sql.DB
+	parsedURL, err := url.Parse(*dbURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Remove database from parsedUrl
+	q := parsedURL.Query()
+	q.Del("database")
+	parsedURL.RawQuery = q.Encode()
+
+	// Open connection to server and create a database.
+	if db, err = sql.Open("cockroach", parsedURL.String()); err != nil {
+		log.Fatal(err)
+	}
+	if _, err = db.Exec("CREATE DATABASE datablocks"); err != nil {
+		// Database might already exist, just log a warning.
+		log.Warning(err)
+	}
+	db.Close()
+
+	// Open connection directly to the new database.
+	q.Add("database", "datablocks")
```

Comment:
`SET DATABASE` and other session-level commands only really make sense for interactive use and in the guts of the driver; I don't think we need a comment each time we choose not to use them. 

We may want to create a default database in `StartTestServer` (or in bootstrapping) so we don't have to repeat this logic every time. It would be nice if examples weren't responsible for creating their own test database, since that's not something that would be present in production code.